### ISSUE TITLE
Relax `1:n` check (false positive)

### ISSUE THREAD

**Issue by Bisaloo** [Bioconductor/BiocCheck#256]
BiocCheck will complain on patterns such as:

```r
1:10
```

But I think this is fine **if the second element is an integer literal**. The danger of `1:` only lies when the second element is a variable is:

```r
1:n
```

I believe having this check lump both cases together is doing a disservice to users as it doesn't help them understand **why and when** `:` is bad.

It also makes BiocCheck unnecessary noisy and teaches users to ignore it, even for a number of important issues currently reported in existing packages.
